### PR TITLE
Update monitor to subscribe to channel

### PR DESCRIPTION
**Pull request by crosbymichael**
Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

File: monitor.go [hunk 1]
```diff
@@ -37,18 +44,33 @@ func (m *defaultMonitor) Run(c *exec.Cmd) error {
...
-	return c.Start()
-}
-
-func (m *defaultMonitor) Wait(c *exec.Cmd) (int, error) {
-	if err := c.Wait(); err != nil {
-		if exitErr, ok := err.(*exec.ExitError); ok {
-			if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
-				return status.ExitStatus(), nil
+func (m *defaultMonitor) Start(c *exec.Cmd) (chan Exit, error) {
+	ec := make(chan Exit, 1)
```
Comment (mlaventure) [inline monitor.go]: nit: this can be moved just before the `go func` to avoid an extra `close(ec)` 2 lines below

File: monitor.go [hunk 2]
```diff
@@ -37,18 +44,33 @@ func (m *defaultMonitor) Run(c *exec.Cmd) error {
...
+				if ws, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+					status = ws.ExitStatus()
+				}
 			}
 		}
-		return -1, err
-	}
-	return 0, nil
+		ec <- Exit{
+			Timestamp: time.Now(),
```
Comment (mlaventure) [inline monitor.go]: nit: get the time as soon as `c.Wait()` returns for more accuracy?

Comment (crosbymichael) [reply monitor.go]: i don't think you will get any closer with the error handling

**Comment by crosbymichael** [comment]
@mlaventure updated

**Comment by crosbymichael** [comment]
@stevvooe ya, this is all internal code for the most part, you can see the full PR for containerd here: https://github.com/containerd/containerd/pull/1452

File: monitor.go (hunk 2 above)
Comment (mlaventure) [reply monitor.go]: Mostly because I don't trust the scheduler :)

Comment (crosbymichael) [reply monitor.go]: its sends the value so the fields are all done

**Pull request merged**